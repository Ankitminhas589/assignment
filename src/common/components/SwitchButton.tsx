import React from 'react';
import {StyleSheet, Switch} from 'react-native';
import {AppColors} from '../utills/theme/Colors';
import {_scaleText} from '../utills/utility';

type IProps = {
  onPress: () => void;
  isActive: boolean;
};
const SwitchButton: React.FC<IProps> = ({onPress, isActive}) => {
  return (
    <Switch
      style={styles.buttonStyle}
      trackColor={{false: AppColors.BUTTON_DISABLED, true: AppColors.APP_GREEN}}
      thumbColor={AppColors.WHITE}
      ios_backgroundColor={isActive ? AppColors.APP_GREEN : AppColors.BUTTON_DISABLED}
      onValueChange={onPress}
      value={isActive}
    />
  );
};

export default SwitchButton;

const styles = StyleSheet.create({
  buttonStyle: {
    transform: [{scaleX: 0.8}, {scaleY: 0.8}],
    marginRight: _scaleText(10).fontSize,
  },
});
