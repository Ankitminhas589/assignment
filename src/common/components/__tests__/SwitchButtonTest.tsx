import React from "react";
import { fireEvent, render } from "@testing-library/react-native";
import SwitchButton from "../SwitchButton";

describe("Image with loader tests", () => {
    it("renders correcly", async () => {
        const { toJSON, getByTestId } = render(<SwitchButton isActive={true} onPress={() => { }} />);
        expect(toJSON()).toMatchSnapshot();
    });
});
