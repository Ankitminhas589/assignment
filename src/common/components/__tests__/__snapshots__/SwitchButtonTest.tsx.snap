// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image with loader tests renders correcly 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#04d167"
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Array [
        Object {
          "marginRight": 20,
          "transform": Array [
            Object {
              "scaleX": 0.8,
            },
            Object {
              "scaleY": 0.8,
            },
          ],
        },
        Object {
          "backgroundColor": "#04d167",
          "borderRadius": 16,
        },
      ],
    ]
  }
  thumbTintColor="#ffffff"
  tintColor="#a7a7a7"
  value={true}
/>
`;
